python/common/org/python/stdlib/datetime/Date.java

```diff
@@ -0,0 +1,304 @@
+package org.python.stdlib.datetime;
+
+import java.util.Collections;
+
+public class Date extends org.python.types.Object {
+
+    @org.python.Attribute
+    public org.python.Object year = __year__();
+
+    @org.python.Attribute
+    public org.python.Object month = __month__();
+
+    @org.python.Attribute
+    public org.python.Object day = __day__();
+
+    @org.python.Attribute
+    public static final org.python.Object min = __min__();
+
+    @org.python.Attribute
+    public static final org.python.Object max = __max__();
+
+    @org.python.Method(__doc__ = "")
+    public Date(org.python.Object[] args, java.util.Map<java.lang.String, org.python.Object> kwargs) {
+
+	super();
+
+	if (args.length + kwargs.size() > 3) {
+	    int val = args.length + kwargs.size();
+	    throw new org.python.exceptions.TypeError("function takes at most 3 arguments (" + val + " given)");
+	}
+
+	if (args.length + kwargs.size() == 3) {
+	    if (kwargs.get("year") != null && args.length == 2) {
+		throw new org.python.exceptions.SyntaxError("positional argument follows keyword argument");
+	    } else if (kwargs.get("month") != null && args.length == 2) {
+		throw new org.python.exceptions.SyntaxError("positional argument follows keyword argument");
+	    }
+	    if (kwargs.get("year") != null) {
+		this.year = kwargs.get("year");
+	    } else {
+		this.year = args[0];
+	    }
+
+	    if (kwargs.get("month") != null) {
+		this.month = kwargs.get("month");
+	    } else {
+		this.month = args[1];
+	    }
+
+	    if (kwargs.get("day") != null) {
+		this.day = kwargs.get("day");
+	    } else {
+		this.day = args[2];
+	    }
+
+	    if ((this.year instanceof org.python.types.Int) && (this.month instanceof org.python.types.Int) && (this.day instanceof org.python.types.Int)) {
+		if (1 <= ((org.python.types.Int) this.year).value && ((org.python.types.Int) this.year).value <= 9999) {
+
+		    if (1 <= ((org.python.types.Int) this.month).value && ((org.python.types.Int) this.month).value <= 12) {
+			if (1 <= ((org.python.types.Int) this.day).value && ((org.python.types.Int) this.day).value <= 31) {
+			} else {
+			    throw new org.python.exceptions.ValueError("day is out of range for month");
+			}
+		    } else {
+			throw new org.python.exceptions.ValueError("month must be in 1..12");
+		    }
+		} else {
+		    throw new org.python.exceptions.ValueError("year " + this.year + " is out of range");
```

Comment:
No need to nest this I think, can check each on their own to make the code more readable